packages/react-storage/src/components/StorageBrowser/store/utils.ts

```diff
@@ -0,0 +1,66 @@
+import { isObject } from '@aws-amplify/ui';
+
+import { LocationData } from '../actions';
+
+import { DEFAULT_STATE } from './constants';
+import {
+  LocationValue,
+  StorageBrowserValue,
+  StoreProviderProps,
+  StoreState,
+} from './types';
+
+export const parseLocationType = ({
+  prefix,
+}: LocationData | LocationValue): LocationData['type'] =>
+  prefix?.endsWith('/') ? 'PREFIX' : prefix?.length ? 'OBJECT' : 'BUCKET';
+
+export function getLocationData(
+  location?: LocationData | LocationValue
+): LocationData | undefined {
+  if (location === undefined) return;
+
+  const { bucket, permissions, prefix, id: _id, type: _type } = location;
+
+  const id = _id ?? crypto.randomUUID();
+  const type = _type ?? parseLocationType(location);
+
+  return { bucket, permissions, prefix, id, type };
+}
+
+export function getState(
+  value: StorageBrowserValue | null | undefined
+): StoreState {
+  if (value === undefined || value === null) return DEFAULT_STATE;
```

Comment:
Nit: Can just use nullish check
```suggestion
  if (value == null) return DEFAULT_STATE;
```

Reply:
Not incorrect but prefer the explicit strict `undefined` and `null` checks for easier grokking